### ISSUE TITLE
cov_pca function reports error if subset is of length 1

### ISSUE THREAD

**Issue by DongyueXie** [stephenslab/mashr#77]
`U.pca = cov_pca(mash_data.L,2,subset=1)`

Error in f %*% d^2 : non-conformable arguments

**Comment by gaow** [comment]
Thanks @DongyueXie . Should we add some minimum sample size check here? It does not make sense to estimate correlation using just one sample ... and it does not make sense either perhaps too small sample size? @stephens999 @pcarbo what do you suggest that we should limit the sample size to (`subset` parameter) for this application? Greater or equals to 3?

**Comment by DongyueXie** [comment]
yeah. I think adding minimum sample size check works. This case is rare but happens.

It happened when i applied mash to kind of 'null' data so subset should be very small.
